### PR TITLE
feat: Rework blog-post-sharebar logic

### PR DESCRIPTION
**Pull request by varholak-peter**

### CODE REVIEW

**Comment by garySiladi** [comment]
some kind of test for the implementation in this component?

**Comment by varholak-peter** [comment]
This line: https://github.com/EconomistDigitalSolutions/fe-blogs/pull/892/files#diff-a42870df44c011f56173c1cdcb37f49bR385
has a condition "true" currently and should be swapped for a check if user is subscriber.

**Comment by varholak-peter** [comment]
Tests will be added soon.

**Comment by garySiladi** [comment]
good 👍 late night tests are the best

File: src/parts/blog-post-sharebar.js [hunk 1]
```diff
@@ -22,91 +22,107 @@ function generateCopyrightUrl(type, title, publicationDate, contentID) {
...
     </a>
   );
+}
+
+/* eslint-disable react/prop-types */
+function createSharebar(props, icons, platform) {
+  const hasPurchaseRights = icons.includes('purchaseRights');
+  if (hasPurchaseRights) {
+    const { type, title, publicationDate, contentID } = props;
+    props.urlOverrides.purchaseRights = generateCopyrightUrl(
```
Comment (tomiberes) [inline src/parts/blog-post-sharebar.js]: You should not modify props.

Comment (varholak-peter) [inline src/parts/blog-post-sharebar.js]: fixed

**Pull request merged**